pages/dao/[symbol]/proposal/new.tsx

```diff
@@ -216,6 +237,53 @@ const New = () => {
     setFormErrors({})
     setForm({ ...form, [propertyName]: value })
   }
+
+  const handleMultiForm = ({ propertyName, value }) => {
+    setMultiFormErrors({})
+    setMultiChoiceForm({ ...multiChoiceForm, [propertyName]: value })
+  }
+
+  const handleNotaButton = () => {
+    const options = multiChoiceForm.options;
+    const last = options.length - 1;
+
+    if (options[last] !== "None of the Above") {
+      options.push("None of the Above");
+      handleMultiForm({propertyName: "options", value: options});
+      setHideNotaButton(true);
+    }
+  }
+
+  const updateOptions = (odx: number, value: string, addOption: boolean)  => {
+    const updatedOptions = multiChoiceForm.options;
+    const nota = "None of the Above";
+    const len = updatedOptions.length-1;
+
+    if (addOption) {
+      if (odx === -1) { // 'add another voting choice' button is clicked
+        if (updatedOptions.length > 9) {
+          return;
+        }
+        if (updatedOptions[len] === nota) {
+          // insert new empty option at the second last position if NOTA exists
+          updatedOptions.splice(len, 0, value);
+        } else {
+          // insert new empty option at the last position if not NOTA not exists
+          updatedOptions.push(value);
```

Comment:
it's not really a good idea to mutate the state like this (which you are doing, you didn't clone this so it's actually the same array).

Instead, I would just do `setMultiChoiceForm(prev => ({ ...prev, options: [...prev.options, ""] }))

Reply:
Corrected.